src/Symfony/Component/Form/ChoiceList/Factory/CachingFactoryDecorator.php

```diff
@@ -174,11 +168,11 @@ public function createListFromLoader(ChoiceLoaderInterface $loader, $value = nul
     /**
      * {@inheritdoc}
      *
-     * @param array|callable|Cache\PreferredChoice|null $preferredChoices The preferred choices
-     * @param callable|false|Cache\ChoiceLabel|null     $label            The option or static option generating the choice labels
-     * @param callable|Cache\ChoiceFieldName|null       $index            The option or static option generating the view indices
-     * @param callable|Cache\GroupBy|null               $groupBy          The option or static option generating the group names
-     * @param array|callable|Cache\ChoiceAttr|null      $attr             The option or static option generating the HTML attributes
+     * @param mixed $preferredChoices
+     * @param mixed $label
+     * @param mixed $index
+     * @param mixed $groupBy
+     * @param mixed $attr
```

Comment:
In https://github.com/symfony/symfony/pull/41998/files#diff-cb1213e76aa5d215066d613f70f20e58f403200646f57a1d82b18983d86d9c3eR171-R178 you added more concrete types. Sure this should be mixed?

Reply:
In that commit, I widened the accepted types to make tests pass, without thinking more about it.

Now, using my brain a bit more, I realized that we has to add `string` because that's what `PropertyAccessDecorator` accepts.
Then I realized: since when is `CachingFactoryDecorator` coupled in anyway with `PropertyAccessDecorator`???

A decorator cannot know restrict the types it accepts here, because it doesn't know what it decorates exactly.

So yes, I'm sure :)

Reply:
Makes sense